### PR TITLE
Fix #2882: Build bazel for util/profile

### LINKED ISSUE #2882: Create a BUILD.bazel file for the profile package in the utility module
This task is part of the build system migration from Gradle to Bazel. 

## Prerequesites

- Make sure you are able to build Oppia using Bazel
- Follow the [Oppia Bazel Setup instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions) to setup Bazel and learn the terminology.
- Read the [Gradle Bazel Migration Best Practies and FAQ](https://github.com/oppia/oppia-android/wiki/Gradle--Bazel-Migration-Best-Practices-and-FAQ) for more context and see if there's an answer there to any question you have.

## Task

1. Create a BUILD.bazel file in

`src/main/java/org/oppia/android/util/profile/`

2. Define the kt_android_library library

Based on the imports in the Kotlin files, add the required third_party dependencies to make the libraries build. 

Run
```
bazel build :all
```
inside the directory, to verify that the libraries build.

Check the Useful links section for a list of third_party dependencies.

You can start with the following stub

```
load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")

kt_android_library(
    name = "directory_management_util",
    srcs = [
        "DirectoryManagementUtil.kt",
    ],
)
```

3. Add the directory to the MIGRATED_PROD_FILES of the module's root BUILD.bazel file

4. Add the libraries to the list of deps dependencies in the targets that require the library

You will have to set the apropriate visbility. See the list of visibilities in the link below.

5. Clean up any unused deps dependencies

6. Verify that the app builds

## Useful links

- [BUILD.bazel files](https://github.com/oppia/oppia-android/search?q=filename%3ABUILD.bazel&type=code)
- [Visibilities](https://github.com/oppia/oppia-android/blob/76dc13dba3ad77aecd2218a5372949d6551f12fe/BUILD.bazel)
- [third_party dependencies](https://github.com/oppia/oppia-android/blob/a70d9b016f0d77dd30c4939041f82ea02ee28d1d/third_party/versions.bzl)

### PR DESCRIPTION
**Pull request by Victor-Titan**
## Explanation

Fixes #2882 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by Victor-Titan** [comment]
@BenHenning @fsharpasharp PTAL

**Review by fsharpasharp — CHANGES_REQUESTED** [review]
Looks good! Added two small comments.

File: utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
```
Comment (fsharpasharp) [inline utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel:5]: Add some documentation to explain what the package does.

File: utility/BUILD.bazel [hunk 2]
```diff
@@ -41,6 +42,7 @@ kt_android_library(
         "//third_party:com_github_bumptech_glide_glide",
         "//third_party:com_google_guava_guava",
         "//third_party:org_jetbrains_kotlinx_kotlinx-coroutines-core",
+        "//utility/src/main/java/org/oppia/android/util/profile:directory_management_util",
```
Comment (fsharpasharp) [inline utility/BUILD.bazel]: We're eventually going to remove this library. If possible, try to add this dependency closer to where it is used.

I see two files that depend on it, not including the test, in the domain and the data modules.

Comment (Victor-Titan) [reply utility/BUILD.bazel]: @fsharpasharp So should I add this in the BUILD.bazel files of the domain and data modules? I think that will make sure no issues occur when the library is removed?

File: utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel (hunk 1 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel:5]: I added a comment declaring what the package does, at the top of the file. Is that enough?

File: utility/BUILD.bazel (hunk 2 above)
Comment (fsharpasharp) [reply utility/BUILD.bazel]: Yes, sounds good and remove directory_management_util from this list of dependencies.

**Comment by Victor-Titan** [comment]
@BenHenning I'm seeing a lot of test failures for maven-dagger and it all points to something regarding jcenter and chaos-pinview. Should we create an issue for this? Seems like another round of flaky tests.

File: utility/BUILD.bazel (hunk 2 above)
Comment (Victor-Titan) [reply utility/BUILD.bazel:45]: I opened the data/BUILD.bazel file and it already had "//utility" in the dependencies. Would you like me to replace that or I should not change anything in the file now?

Comment (Victor-Titan) [reply utility/BUILD.bazel]: @fsharpasharp PTAL now, I've done the necessary changes. I forgot to tag you.

File: utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,16 @@
+"""
+General purposes utilities to manage Directories.
```
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel]:
```suggestion
General purposes utilities to manage directories.
```

No need to capitalize here.

**Review by BenHenning — APPROVED** [review]
Thanks @Victor-Titan! Just one nit comment, otherwise this LGTM.

File: utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel (hunk 3 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/profile/BUILD.bazel]: @BenHenning Ok I removed the capitalization.

**Comment by Victor-Titan** [comment]
@fsharpasharp PTAL

**Review by fsharpasharp — APPROVED** [review]

**Comment by BenHenning** [comment]
Ah @Victor-Titan sorry I missed your earlier issue. JCenter was having some outages a few days ago, but I did file #2923 to eventually move off of it.

**Review by BenHenning — APPROVED** [review]
Thanks @Victor-Titan! LGTM.

**Comment by BenHenning** [comment]
Gradle failure looks like #2844. Restarting & enabling auto-merge.

**Pull request merged**